src/index.test.js

```diff
@@ -487,6 +487,27 @@ describe("links", () => {
     });
   });
 
+  it("accepts `as` prop to render custom element", () => {
+    let ref;
+    let div = document.createElement("div");
+    let className = "btn btn-link";
+    let LinkComponent = React.forwardRef(({ children, ...props }, ref) => (
+      <a className={className} ref={ref} {...props}>
+        {children}
+      </a>
+    ));
+    ReactDOM.render(
+      <Link to="/" as={LinkComponent} ref={node => (ref = node)} />,
+      div
+    );
+    try {
+      expect(ref).toBeInstanceOf(HTMLAnchorElement);
+      expect(ref.className).toBe(className);
```

Comment:
these assertions look like they would pass even without this `as={LinkComponent}` because the default implementation is to render an Anchor tag.

Reply:
First one, yeah. Second one, no.

Reply:
I haven't looked close yet, I guess I assumed the className got forwarded on. but yeah, it would be nice to tighten up the test to ensure it's not going to be broken later.

Reply:
You are right 🤦‍♂ Quick fix would be to use a button element instead.

I'll think on broader accessibility concerns. I'm not sure there is a good solution to that. It feels like a reasonable feature, but agreed we don't want to enable easy foot guns (i.e. actually using a button like I'm describing 😅)